src/alloc/isolated_alloc.rs

```diff
@@ -0,0 +1,332 @@
+use std::alloc::{self, Layout};
+use std::sync;
+
+static ALLOCATOR: sync::Mutex<IsolatedAlloc> = sync::Mutex::new(IsolatedAlloc::empty());
```

Comment:
I was hoping we wouldn't need this any more and could instead store this in the `Machine` and pass it to the allocation via the `params`... otherwise, the allocations of multiple native-libs machines could still get mixed here.

But that's harder now that parameters are their own associated types. It should still be possible though, AFAIK lifetime generic associated types are stable?

Reply:
Ended up going with a hashmap, but I can change that if you think it's not the right call. I just couldn't see a non-intrusive way to do it outside of either (a) passing a raw pointer instead of a reference (yikes) or (b) using something like `Arc<RefCell<IsolatedAllocator>>` which I'm not sure is worth the overhead and runtime panic risk

Reply:
I think I found a way to make the Rc-only thing work so I'll do that! + it might fix other stuff that's currently bothering me, lol

Reply:
Ah I guess now I nerd-sniped you into making the many-seeds thing works for native-libs mode... sorry for that.^^

Just for the allocator, you don't need an ID -- you can entirely avoid even having a global variable here and just store another `Rc<...>` pointing to the allocator in the machine itself. That would only hold the allocations for that one machine so it can easily list "its" pages (and it also neatly involves having concurrent machines content on the same lock). But I don't know which further constraints are imposed by the ptrace thing.

Reply:
Sorry, I mean many-seeds can coexist but only one can be in "ffi mode" at a time, with the others blocking until that one finishes. Otherwise it becomes borderline impossible to trace properly, and perhaps actually impossible when it comes to e.g. intercepting syscalls